src/test/java/ladder/MockRungGenerator.java

```diff
@@ -0,0 +1,23 @@
+package ladder;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import ladder.domain.Rung;
+import ladder.domain.randomGenerator.RungGenerator;
+
+public class MockRungGenerator implements RungGenerator {
```

Comment:
https://martinfowler.com/bliki/TestDouble.html
https://martinfowler.com/articles/mocksArentStubs.html
요 자료들을 보셨나요? (국내 블로그들의 레퍼런스를 타고타고타고 들어가다보면 나오는 원전입니다)
사실 Mock 이라고 하면 모든 테스트 더블을 아우르는 개념을 생각하는게 대부분입니다
Fake, Spy, Stub 같은 디테일은 Mocking 라이브러리를 사용할 때 정도에야 논의되는거 같네요

어쨌든 목 객체를 사용하는 경우는 2가지 입니다

1. 외부의존성 (DB / API콜) 등이 있어서 실제 객체를 사용하기 곤란한 경우
2. 지금처럼 테스트 세팅이 어려운 경우

우선순위에 대해서 말씀해주셨는데, 테스트하기 어려운 경우를 인터페이스와 구현을 이용해 풀어냈다는게 중요하지 mock이나 stub이나 fake냐 구분해 학습하는 건 중요하지 않다고 생각해요. 

'하지만 빨랐죠?' 밈처럼, 문제를 해결했다는 게 중요하지 않을까요

\2. private를 확인하기 위해 트릭을 사용했다는 말씀은 mock 객체를 이용한 걸 말씀하시는 걸까요?
맞다면 좋은 해결 방법이라고 생각합니다. 요 방식 말고 Random을 테스트에서 분리해내는게 불가능하지 않나? 생각 드네요
RungGenerator의 메서드가 많았다면 랜덤 기능 말고는 최소한으로 mocking하도록 extends 했을거 같아요

```suggestion
public class MockRungGenerator extends RandomRungGenerator {

... // 다른 메서드들은 내버려둠

    @Override
    public Rung getRandomStatusRung() {
        if (!mockedRungs.hasNext()) {
            throw new NoSuchElementException("다음 Mock 데이터를 찾을 수 없습니다.");
        }
        return mockedRungs.next();
    }
...
}
```

Reply:
첨부해주신 사이트는 인지하지 못하고 있었는데 또 하나의 오아시스를 발견한 듯한 느낌이에요!!
마틴 파울러의 포스트들을 시간 날 때마다 하나하나 읽어봐야겠어요 정말 감사합니다 😊 

>RungGenerator의 메서드가 많았다면 랜덤 기능 말고는 최소한으로 mocking하도록 extends 했을거 같아요
오.. 새로운 깨달음입니다. 참고하겠습니다!